### ISSUE TITLE
Bug explanation

### ISSUE THREAD

**Issue by SeanPercy** [SeanPercy/prisma-assignment#2]
tldr: Objects being returned by `subscription.next()` are supposed to be of type `IteratorResult<UserSubscriptionPayload>`. Therefore, their `value` fields should contain objects of type 
`UserSubscriptionPayload`. However, the `value` fields of the actual objects being returned by `subscription.next()` do not contain objects of `UserSubscriptionPayload`. Maybe auto-generation is wrongly configured.

------
In order to properly follow through these steps, you have to comment out line 39 in `main.ts` before running `npm run dev`. Once the app has launched open `http://localhost:3000/playground` 

<img width="942" alt="7" src="https://user-images.githubusercontent.com/7240150/51676239-57528900-1fd6-11e9-9834-c544c20717ca.png">

and run a mutation like the following:

```graphql
mutation {
  createUser(data: {username: "a", password: "a") {
    usernanme
    password
  }
}
```

Now check the console output of your IDE and the `main.ts` file

<img width="742" alt="1" src="https://user-images.githubusercontent.com/7240150/51674311-92ea5480-1fd0-11e9-9a55-8fa15180b878.png">

As we can see above, in line 27, a variable named `result` gets declared and is assigned to the call of the method `subscription.next()`, which is supposed to a return an object that matches the interface of type `IteratorResult<UserSubscriptionPayload>`. The interface definition can be found inside `./generated/prisma.ts`

It should be shaped like this:
```javascript
{
  value: UserSubscriptionPayload, 
  done: boolean
}
```

`UserSubscriptionPayload` itself is also an auto-generated interface inside `./generated/prisma.ts` and looks like this:
 
```javascript
{
  mutation: MutationType
  node?: User | null
  updatedFields?: Array<String> | null
  previousValues?: UserPreviousValues | null
}
```

In line 28, logging the variable `result` to the console generates the following output: 
```javascript
{
  value:
    { 
      user: 
        { 
          mutation: 'CREATED',
          node: [Object],
          updatedFields: null 
        } 
    },
  done: false 
}
```

However, according to the `prisma.ts` it should look like this instead:
```javascript
{ 
  value:
    {
      mutation: MutationType
      node?: User | null
      updatedFields?: Array<String> | null
      previousValues?: UserPreviousValues | null
    }, 
  done: false 
}
```

Since the field `user` is not defined on `UserSubscriptionPayload`, trying to access `result.value.user` causes the TypeScript compiler to throw an error (`TS2339: Property 'user' does not exist on type 'UserSubscriptionPayload'`), as it can be seen in the next screenshot.

<img width="806" alt="4b" src="https://user-images.githubusercontent.com/7240150/51676154-165a7480-1fd6-11e9-94b5-04bd2c5fb4d1.png">

Therefore, the field `user` and all fields nested inside it become inaccessible, unless performing a type cast of `result` to type `any` , as we can see in the two screenschots below  
  
<img width="715" alt="5" src="https://user-images.githubusercontent.com/7240150/51675389-d5616080-1fd3-11e9-9f4c-0deb673866f3.png">

<img width="718" alt="6" src="https://user-images.githubusercontent.com/7240150/51675413-e7db9a00-1fd3-11e9-98f0-08b163926fc9.png">

In order to fix this, either the generation of the shape of `UserSubscriptionPayload` has to be changed, so that the resulting shape matches the shape of the `value` field of the objects being returned by `subscription.next()` OR the shape of the objects that are actually being returned by `subscription.next()` has to be changed in a way, that its `value` field matches the shape of objects of type `UserSubscriptionPayload`
